From 408a6bc16a011bf23df009891d03bb15c18c31c7 Mon Sep 17 00:00:00 2001
From: Marsell Kukuljevic <marsell@joyent.com>
Date: Wed, 10 Apr 2019 23:13:34 +0200
Subject: [PATCH] TRITON-1388 Remove process.exit() from dapi lib Reviewed by:
 Trent Mick <trentm@gmail.com> Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
 Approved by: Josh Wilsdon <jwilsdon@joyent.com>

---
 lib/allocator.js | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/lib/allocator.js b/lib/allocator.js
index d07b872..53a6571 100644
--- a/lib/allocator.js
+++ b/lib/allocator.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2015, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -475,10 +475,11 @@ Allocator.prototype._createExpression = function (description, algorithmLookup)
 	var self = this;
 	var expression = [];
 
-	var errExit = function (errMsg) {
-		self.log.error('Bad expression given: ' + errMsg);
-		process.exit(1);
-	};
+	function errExit(errStr) {
+		var errMsg = 'Bad expression given: ' + errStr;
+		self.log.error(errMsg);
+		throw new Error(errMsg);
+	}
 
 	var command = description[0];
 
-- 
2.21.0

